lgogdownloader fails to build

**Describe the bug**
lgogdownloader fails to build on unstable. The configure phase breaks with error
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/build/source/JSONCPP_INCLUDE_DIR
   used as include directory in directory /build/source

-- Configuring incomplete, errors occurred!
```

**To Reproduce**
Steps to reproduce the behavior:
1. Build lgogdownloader on master/unstable

**Additional context**
Caused by the following commit:
```
256a3735ec776680341ecfff0ce891d056532dec is the first bad commit
commit 256a3735ec776680341ecfff0ce891d056532dec
Author: R. RyanTM <ryantm-bot@ryantm.com>
Date:   Wed Nov 27 17:47:51 2019 -0800

    jsoncpp: 1.9.1 -> 1.9.2

 pkgs/development/libraries/jsoncpp/default.nix | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
```

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.87, NixOS, 20.03pre204199.3140fa89c51 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(alex): `""`
 - channels(root): `"nixos-20.03pre205624.8636580d6f4, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/development/libraries/jsoncpp/default.nix
+++ b/pkgs/development/libraries/jsoncpp/default.nix
@@ -1,4 +1,4 @@
-{ stdenv , fetchFromGitHub , cmake , python }:
+{ stdenv, fetchFromGitHub, cmake, python, fetchpatch }:
 
 stdenv.mkDerivation rec {
   pname = "jsoncpp";
@@ -29,6 +29,14 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ cmake python ];
+
+  # fix inverted sense in isAnyCharRequiredQuoting on aarch64. See: https://github.com/open-source-parsers/jsoncpp/pull/1120
+  patches = stdenv.lib.optionals stdenv.isAarch64 [
+    (fetchpatch {
+      url = "https://github.com/open-source-parsers/jsoncpp/commit/9093358efae9e5981aa60013487fc7215f040a59.patch";
+      sha256 = "1wiqp70sck2md14sfc0zdkblqk9750cl55ykf9d6b9vs1ifzzzq5";
+     })
+  ];
 
   cmakeFlags = [
     "-DBUILD_SHARED_LIBS=ON"

--- a/pkgs/games/lgogdownloader/default.nix
+++ b/pkgs/games/lgogdownloader/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchFromGitHub, cmake, pkgconfig, curl, boost, liboauth, jsoncpp
-, htmlcxx, rhash, tinyxml-2, help2man }:
+, htmlcxx, rhash, tinyxml-2, help2man, fetchpatch }:
 
 stdenv.mkDerivation rec {
   pname = "lgogdownloader";
@@ -15,6 +15,14 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ cmake pkgconfig help2man ];
 
   buildInputs = [ curl boost liboauth jsoncpp htmlcxx rhash tinyxml-2 ];
+
+  patches = [
+    # Fix find_path for newer jsoncpp. Remove with the next release
+    (fetchpatch {
+      url = "https://github.com/Sude-/lgogdownloader/commit/ff353126ecda61824cf866d3807c9ebada96282e.patch";
+      sha256 = "1xr1lwxlrncrj662s9l1is1x1mhs1jbwlj8qafixz5hw2kx22w19";
+    })
+  ];
 
   meta = {
     homepage = https://github.com/Sude-/lgogdownloader;

